cs/ObjectPool.cs

```diff
@@ -157,6 +160,8 @@ public void UpdateFactory(ObjectFactory<TSource, TObject> factory)
 
         internal void ReturnObject(PooledObject<TSource, TObject> pooledObject)
         {
+            Contract.Ensures(pooledObject != null);
```

Comment:
why is this better than manually throwing? I understand that code doesn't have to deal with contract violations, but because of the silent failures, won't it be hard to debug?